docs/guides/js-waku/store-retrieve-messages.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Retrieve Messages Using Store
+---
+
+This guide provides detailed steps to create a light node for retrieving and filtering historical messages using the [Store protocol](/overview/concepts/protocols#store).
+
+## Create a Light Node
+
+Use the `createLightNode()` function to create a light node and interact with the Waku Network:
+
+```js
+import { createLightNode } from "@waku/sdk";
+
+// Create and start a light node
+const node = await createLightNode({ defaultBootstrap: true });
+await node.start();
+```
+
+## Connect to Store Peers
+
+Use the `waitForRemotePeer()` function to wait for the node to connect with store peers:
+
+```js
+import { waitForRemotePeer, Protocols } from "@waku/sdk";
+
+// Wait for a successful peer connection
+await waitForRemotePeer(node, [Protocols.Store]);
+```
+
+## Choose a Content Topic
+
+[Choose a content topic](/overview/concepts/content-topics) for filtering the messages to retrieve and create a `decoder` for [message decryption](https://rfc.vac.dev/spec/26/):
+
+```js
+import { createDecoder } from "@waku/sdk";
+
+// Choose a content topic
+const contentTopic = "/store-guide/1/message/proto";
+
+// Create a message decoder
+const decoder = createDecoder(contentTopic);
+```
+
+## Retrieve Messages
+
+`js-waku` provides the `queryOrderedCallback()` and `queryGenerator()` functions for querying `Store` nodes and retrieving historical or missed messages. The responses from `Store` nodes are paginated and require you to handle them sequentially, processing each page when received.
```

Comment:
... are paginate and requires you to handle each page sequentially (as network requests for each page are done in sequence).

Something like that.The messages in a given page do not have to be handled sequentially.